From dc6d13ff95a8dd4f81a39b2c636e9f2ec255e557 Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Fri, 13 Nov 2015 10:48:22 -0600
Subject: [PATCH 0732/5242] MLK-11939: ARM: dts: Add ddrsmp to device tree for
 i.mx6sx sabreauto and sdb board

commit  02533a57a2dc835f4bf50e48c0607a17a6156629 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add ddrsmp parameter to device tree for i.MX6SX Sabreauto/Sabresd board.

DDRSMP value:
Signed-off-by: Meng Li <Meng.Li@windriver.com>
0 ---- i.MX6SX Sabresd board (RevB and RevA)
2 ---- i.MX6SX Sabreauto board

The Sabresd RevA board also needs to reduce clock to 29Mhz according to
the Spansion spec.

Signed-off-by: Han Xu <b45815@freescale.com>
---
 arch/arm/boot/dts/imx6sx-sabreauto.dts |    1 +
 arch/arm/boot/dts/imx6sx-sdb-reva.dts  |    5 +++--
 arch/arm/boot/dts/imx6sx-sdb.dts       |    2 ++
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-sabreauto.dts b/arch/arm/boot/dts/imx6sx-sabreauto.dts
index 8e816cc..b801083 100644
--- a/arch/arm/boot/dts/imx6sx-sabreauto.dts
+++ b/arch/arm/boot/dts/imx6sx-sabreauto.dts
@@ -375,6 +375,7 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_qspi1_1>;
 	status = "okay";
+	ddrsmp=<2>;
 
 	flash0: n25q256a@0 {
 		#address-cells = <1>;
diff --git a/arch/arm/boot/dts/imx6sx-sdb-reva.dts b/arch/arm/boot/dts/imx6sx-sdb-reva.dts
index 5dbf130..98a88f6 100644
--- a/arch/arm/boot/dts/imx6sx-sdb-reva.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb-reva.dts
@@ -152,13 +152,14 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_qspi2>;
 	status = "okay";
+	ddrsmp=<0>;
 
 	flash0: s25fl128s@0 {
 		reg = <0>;
 		#address-cells = <1>;
 		#size-cells = <1>;
 		compatible = "spansion,s25fl128s", "jedec,spi-nor";
-		spi-max-frequency = <66000000>;
+		spi-max-frequency = <29000000>;
 	};
 
 	flash1: s25fl128s@1 {
@@ -166,6 +167,6 @@
 		#address-cells = <1>;
 		#size-cells = <1>;
 		compatible = "spansion,s25fl128s", "jedec,spi-nor";
-		spi-max-frequency = <66000000>;
+		spi-max-frequency = <29000000>;
 	};
 };
diff --git a/arch/arm/boot/dts/imx6sx-sdb.dts b/arch/arm/boot/dts/imx6sx-sdb.dts
index fa2cbdc..9e98f62 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb.dts
@@ -35,6 +35,8 @@
 	status = "okay";
 
 #ifndef SPANSIONFLASH
+	ddrsmp=<0>;
+
 	flash0: n25q256a@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
-- 
1.7.9.5

